### PR TITLE
Convert compoent  kubeedge to edgeruntime

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <zhu733756@kubesphere.io>

### What type of PR is this?

/kind feature
/kind api-change

### What this PR does / why we need it:

To support more edge frameworks as optional.  The former component kubeedge needs to be converted to EdgeRuntime. Also, the proxy service has a change:
```
edgeruntime:
      endpoint: http://edgeservice.kubeedge.svc/api/
```

See #https://github.com/kubesphere/ks-installer/pull/1829 for a detailed introduction.

### Special notes for reviewers:
```
None
```

### Does this PR introduced a user-facing change?

```release-note
KubeEdge proxy service changes to http://edgeservice.kubeedge.svc/api/
```

As for console, API group changes from kubeedge.kubesphere.io to edgeruntime.kubesphere.io, no other breaks.

### CODE REVIEW

**Comment by zhu733756** [comment]
Hi, guys, after this pr merged, you can bring your own computing framework, call #4170 #4153

File: pkg/simple/client/edgeruntime/options.go [hunk 1]
```diff
@@ -43,6 +43,6 @@ func (o *Options) Validate() []error {
 }
 
 func (o *Options) AddFlags(fs *pflag.FlagSet, c *Options) {
-	fs.StringVar(&o.Endpoint, "edge-watcher-endpoint", c.Endpoint,
```
Comment (zhu733756) [inline pkg/simple/client/edgeruntime/options.go:46]: maybe need to add a namespace flag.

File: pkg/apiserver/config/config_test.go [hunk 2]
```diff
@@ -177,8 +177,8 @@ func newTestConfig() (*Config, error) {
 			IndexPrefix: "ks-logstash-auditing",
 			Version:     "6",
 		},
-		KubeEdgeOptions: &kubeedge.Options{
-			Endpoint: "http://edge-watcher.kubeedge.svc/api/",
```
Comment (zhu733756) [inline pkg/apiserver/config/config_test.go:181]: should be configured as a namespace pramameter

**Comment by zryfish** [comment]
Does this PR introduce API breaking change?

**Comment by zhu733756** [comment]
> Does this PR introduce API breaking change?

Yeah, see the related link in ks-installer  #kubesphere/ks-installer#1829 .  As to console, group change from kubeedge.kubesphere.io to edgeruntime.kubesphere.io, no other breaks.

**Comment by zryfish** [comment]
That's big, I suggest keeping both API groups at the moment, mark `kubeedge.kubesphere.io` as deprecated, and remove it in the later version.  Always be careful about API break changes.

**Comment by zhu733756** [comment]
> That's big, I suggest keeping both API groups at the moment, mark `kubeedge.kubesphere.io` as deprecated, and remove it in the later version. Always be careful about API break changes.

All right. Will fix it by another commit later.

**Comment by benjaminhuo** [comment]
Make sure you handle the deprecation well in the edgeruntime service

**Comment by zhu733756** [comment]
> Make sure you handle the deprecation well in the edgeruntime service because the old API and the new API will all be proxied to the backend edge runtime service

Just make a temp fixed.

**Comment by zhu733756** [comment]
> Make sure you handle the deprecation well in the edgeruntime service because the old API and the new API will all be proxied to the backend edge runtime service

Have tested the method in the restful-go framework named Deprecate(), but seems to make no effect.  See https://pkg.go.dev/github.com/emicklei/go-restful#RouteBuilder.Deprecate.

**Pull request merged**